src/utils/activity-object-helper.ts

```diff
@@ -15,12 +16,20 @@ export class ActivityObjectHelper {
     object: ActivityObject | Link | URL | Array<ActivityObject | URL>
   ): string | undefined {
     const lang = navigator.language.substr(0, 2);
-    if (ActivityObjectHelper.hasProperty(object, "name")) {
+    if (ActivityObjectHelper.hasProperty(object, "name") && object.name) {
       return (object as ActivityObject).name;
     } else if (ActivityObjectHelper.hasProperty(object, "nameMap")) {
-      return (object as RdfLangString).nameMap[lang];
-    } else if (ActivityObjectHelper.hasProperty(object, "map")) {
-      return (object as RdfLangString).map[lang];
+      const rdfLangStringObject = object as RdfLangString;
+
+      if (rdfLangStringObject.nameMap) {
+        return rdfLangStringObject.nameMap[lang];
+      }
+    } else if (ActivityObjectHelper.hasProperty(object, "preferredUsername")) {
```

Comment:
I would use preferredUsername as the first condition

Reply:
I understood the standard that way that the `name` property is kind of the "display name" and preferredUsername is more the system name of the user. (see twitter where you have "@PascalMySelf" but my display name might just be "Pascal"). What do you think?

Reply:
You're right. Mastadon does the same